docs/synapse/adminguide.rst

```diff
@@ -51,4 +51,67 @@ this process, the layer will be readable but writes will hang due to needing to
 guarantee that edits are immediately observable like a normal layer.  During that process, you may track progress by calling
 the ``getMirrorStatus()`` API on the ``storm:layer`` object within the Storm runtime.
 
+Add Extended Model Elements
+---------------------------
+
+The Synapse data model in a Cortex can be extended with custom forms, properties and tag
+properties by using the model extension Storm Library (:ref:`stormlibs-lib-model-ext`). Extended model
+forms and properties must have names beginning with an underscore (``_``) to avoid potential
+conflicts with built-in model elements. Tag properties are not part of the base Synapse
+data model and are not required to begin with an underscore.
+
+When adding a form, ``$lib.model.ext.addForm`` takes the following arguments:
+
+``formname``
+    Name of the form, must begin with an underscore (``_``) and contain at least one colon (``:``).
+
+``basetype``
+    The `Synapse data model type`_ for the form.
+
+``typeopts``
+    A dictionary of type specific options.
+
+``typeinfo``
+    A dictionary of info values for the form.
+
+To add a new form named ``_foocorp:name``, which contains string values which will be
+normalized to lowercase, with whitespace stripped from the beginning/end::
+
+    $lib.model.ext.addForm(_foocorp:name, str, ({'lower': $lib.true, 'strip': $lib.true}), ({'doc': 'Foocorp name'}))
+
+If the form is no longer in use and there are no nodes of this form in the Cortex, it can be removed with::
+
+    $lib.model.ext.delForm(_foocorp:name)
+
+When adding properties, ``$lib.model.ext.addFormProp`` takes the following arguments:
+
+``formname``
+    Name of the form to add the property to, may be a built-in or extended model form.
+
+``propname``
+    Relative name of the property, must begin with an underscore (``_``).
+
+``typedef``
+    A tuple of (``type``, ``typeopts``) which defines the type for the property
+
+``propinfo``
+    A dictionary of info values for the property.
+
+To add a property named ``_score`` to the ``_foocorp:name`` form which contains
```

Comment:
Please include an example of adding a property to an existing form and explicitly state that you can do that :)

Reply:
I'll add an example, but it does already say that formname may be a built-in or extended model form